apps/definition/lib/definition.ex

```diff
@@ -36,7 +36,12 @@ defmodule Definition do
 
       @impl Definition
       def new(%{} = input) do
-        map = for {key, val} <- input, do: {:"#{key}", val}, into: %{}
+        map =
+          for {key, val} <- input,
+              do: {:"#{key}", val},
+              into:
+                %{}
+                |> Map.put_new_lazy(:id, fn -> UUID.uuid4() end)
```

Comment:
This is hard to read. The `for` loop executes completely before the result is piped to your `Map.put_new_lazy` call. It took me a minute to parse that. Elixir's `for` loops are only good as long as they can sit on one line (IMO). After that, you're better off using `Enum`.

```elixir
map =
  Enum.map(input, fn {key, val} -> {:"#{key}", val} end)
  |> Map.new()
  |> Map.put_new_lazy(:id, &UUID.uuid4/0)
```

Reply:
FWIW the for loop used to be on one line, but the code formatter suggested the multiline format. I've changed it to your Enum.map function.